R/functions_enrichment.R

```diff
@@ -0,0 +1,36 @@
+#' Perform Over-Representation Analysis (ORA) for a single cluster
+#'
+#' This function performs Over-Representation Analysis (ORA) for a given set of 
+#' genes within a cluster. It uses the clusterProfiler package to identify 
+#' enriched biological processes or pathways.
+#'
+#' @param genes A vector of gene symbols to perform ORA on.
+#' @param universe A vector of all genes in the universe of interest.
+#' @param cluster The cluster identifier for which ORA is being performed.
+#' @param term2gene A data frame mapping gene sets (terms) to their corresponding genes.
+#' 
+#' @import clusterProfiler
+#' 
+#' @return A data frame containing the results of the ORA, including the cluster identifier.
+perform_ora <- function(genes, universe, cluster, term2gene) {
```

Comment:
- Comment on existing enrichR functions to say that they are kept but not used at the moment
- Why do we need this function?
  - Seems like it "only" calls clusterProfiler::enricher
  - Cutoffs as parameters
  - Assign cluster in the main code so the function is more generic

Reply:
I do not see enrichR being called in my function.

Reply:
I created another function to make the call to enricher because this way I can handle any use cases without disturbing the main code.
What do you mean by assign cluster in the main code?